controllers/installation_controller.go

```diff
@@ -630,7 +631,17 @@ func (r *InstallationReconciler) ReconcileHelmCharts(ctx context.Context, in *v1
 		return nil
 	}
 
-	combinedConfigs := mergeHelmConfigs(ctx, meta, in)
+	isHAEnabled, err := registry.IsHAEnabled(ctx, in, r.Client)
+	if err != nil {
+		return fmt.Errorf("failed to check if HA is enabled: %w", err)
+	}
+
+	err = registry.EnsureSecrets(ctx, meta, r.Client)
+	if err != nil {
+		return fmt.Errorf("failed to ensure registry secrets: %w", err)
+	}
```

Comment:
Could we run these outside of `ReconcileHelmCharts`? This function is frankly already too large